### PR TITLE
fix for include-referenced-projects flag

### LINKED ISSUE #1848: include-referenced-projects lead to strange changes in nuspec
I have two projects ProjA, ProjB. 
Both have paket.template files. 
ProjA have reference to ProjB (in .csproj file)

I execute `paket pack output test` in solution folder 
In result package ProjA have a dependence with min version specified to package ProjB.

But if i execute `paket pack output test include-referenced-projects`
result package ProjA have a dependence without specified version to package ProjB
(`pin-project-references` do not change situation in this case)

Expected behavior: ProjA have a dependence with min version to ProjB (or with fixed version if `pin-project-references` was specified)

Sample solution in attached archive
[PackIncludeRefProj.zip](https://github.com/fsprojects/Paket/files/402862/PackIncludeRefProj.zip)

### PR DESCRIPTION
**Pull request by lexarchik**
Try to fix #1848

### CODE REVIEW

File: src/Paket.Core/Packaging/PackageMetaData.fs [hunk 1]
```diff
@@ -296,30 +296,20 @@ let findDependencies (dependenciesFile : DependenciesFile) config platform (temp
                 |> Seq.map (fun kv -> kv.Value.NugetPackages |> List.map (fun p -> Some kv.Key, p, None))
                 |> List.concat
             | None -> []
-          
+
         [if includeReferencedProjects then
             for proj in project.GetAllReferencedProjects(false,projDeps) |> Seq.filter ((<>) project) do
-                match proj.FindTemplatesFile() with
-                | Some templateFileName when TemplateFile.IsProjectType templateFileName ->
-                    match TemplateFile.Load(templateFileName, lockFile, None, Seq.empty |> Map.ofSeq).Contents with
```
Comment (lexarchik) [inline src/Paket.Core/Packaging/PackageMetaData.fs]: The idea of this fix is here: we can't just Load template file for referenced project - we must merge it with info from project (we do so in merge method in PackageProcess.fs).

File: src/Paket.Core/Packaging/PackageProcess.fs [hunk 2]
```diff
@@ -148,41 +148,38 @@ let Pack(workingDir,dependenciesFile : DependenciesFile, packageOutputPath, buil
...
-        | Some template -> 
-            let projects = getAllProjectsFiles (FileInfo(template).Directory.FullName)
-            projects
-            |> Map.filter (fun p (t,_) -> normalizePath (Path.GetFullPath t.FileName) = normalizePath (Path.GetFullPath template))
-        | None -> getAllProjectsFiles workingDir
+                | Some file -> normalizePath (Path.GetFullPath file) = normalizePath (Path.GetFullPath merged.FileName)
+                | None -> true
+            
+            Path.GetFullPath projectFile.FileName |> normalizePath,(merged,projectFile,willBePacked))
+        |> Map.ofArray
```
Comment (lexarchik) [inline src/Paket.Core/Packaging/PackageProcess.fs]: When we pack single template findDependencies still need all project-template info for choose right version of nuget dependencies if flag --include-referenced-projects was provided.

File: src/Paket.Core/Packaging/PackageMetaData.fs [hunk 3]
```diff
@@ -159,7 +159,7 @@ let addFile (source : string) (target : string) (templateFile : TemplateFile) =
     | IncompleteTemplate -> 
         failwith (sprintf "You should only try and add files to template files with complete metadata.%sFile: %s" Environment.NewLine templateFile.FileName)
 
-let findDependencies (dependenciesFile : DependenciesFile) config platform (template : TemplateFile) (project : ProjectFile) lockDependencies minimumFromLockFile pinProjectReferences (map : Map<string, TemplateFile * ProjectFile>) includeReferencedProjects (version :SemVerInfo option) specificVersions (projDeps) =
+let findDependencies (dependenciesFile : DependenciesFile) config platform (template : TemplateFile) (project : ProjectFile) lockDependencies minimumFromLockFile pinProjectReferences (projectWithTemplates : Map<string, TemplateFile * ProjectFile * bool>) includeReferencedProjects (version :SemVerInfo option) specificVersions (projDeps) =
```
Comment (matthid) [inline src/Paket.Core/Packaging/PackageMetaData.fs]: I prefer to introduce records instead of triples..

Comment (lexarchik) [reply src/Paket.Core/Packaging/PackageMetaData.fs]: i will try to change it next time:)

**Pull request merged**